### ISSUE TITLE
is this possible to implement protopnet on cifar 10

### ISSUE THREAD

**Issue by javadrastegari** [fanconic/this-does-not-look-like-that#4]
hi, 
i was wondering if it would be possible to implement protopnet on cifar 10?
thank you

**Comment by fanconic** [comment]
Hi there,

Yes, you can implement ProtoPNet pretty much on any computer vision dataset. However, you might want to change the encoder architecture, as a deep network like ResNet50 might be overparametrized for CIFAR10.

**Comment by javadrastegari** [comment]
hi again 
thanks for you response 
i implemented protopnet on cifar10 based on your code, and  accuracy  is about 80% on cifar 10 test set.
but the problem is the prototype that extracted from train set are the same. 
for example for dog class all 10 prototype are the same.
any idea why?
thanks a lot for your help

**Comment by fanconic** [comment]
This is an interesting phenomenon that happens, when training ProtoPNets. This is mainly when your prototypes are all converging around a single projection.
From my understanding, there are ways how to address this in the loss function, by adding a term in the loss function that penalizes when the prototypes are too close too each other.
I think you can find something interesting here: https://arxiv.org/abs/2110.02058

**Comment by javadrastegari** [comment]
that is interesting idea 
actually i solved this issue by increasing start point of push_up_dataset.
by default push_up_dataset start in epoch 10 i change it to 70 so prototype will be created  in epoch 70 and this solved the issue.
another problem is that prototypes in CUB are portion of original Image. but in CIFAR10 prototypes are as same size as original image (actually prototypes are original image without any change) any idea why?
maybe this is why i cant get accuracy more than 80% in cifar10 test set. 
anyway i think it is interesting that no one implemented protopnet on cifar10 or any 32 *32 dataset (as far as i know ) despite ProtoPnet citation.
thanks for your advice

**Comment by fanconic** [comment]
This is most likely because of the projection from the latent space to the input space. It looks that your field-of-view when projecting back is the whole picture, and 95% of the activated values fall on all the pixels. You could potentially try to use a smaller encoder network.

**Comment by javadrastegari** [comment]
sorry but i dont understand, what do you mean by "use a smaller encoder network"
could you please explain which part of your code is about encoder network?

**Comment by fanconic** [comment]
The encoder, is the part of the network that reduces the image down to latent space. In the our case it was either the ResNet18, ResNet34, or VGG-19. These networks might be too large for 32*32 images. So you can try to use smaller networks.

However, all your questions are more specific to ProtoPNet itself, rather than our experiments to show the shortcommings of the interpretability mechanism. You might get better answers from the original authors on their github page: https://github.com/cfchen-duke/ProtoPNet
